addons/account/models/account_move.py

```diff
@@ -1842,30 +1855,31 @@ def _compute_partial_lines(self):
             #when running the manual reconciliation wizard, don't check the partials separately for full
             #reconciliation or exchange rate because it is handled manually after the whole processing
             return self
-
-        # check if the reconcilation is full
-        # Determine the main foreign currency across all past partial reconciliations
-        currency = None
-
+        #check if the reconcilation is full
+        #first, gather all journal items involved in the reconciliation just created
         aml_set = aml_to_balance = self.env['account.move.line']
         total_debit = 0
         total_credit = 0
         total_amount_currency = 0
+        #make sure that all partial reconciliations share the same secondary currency otherwise it's not
+        #possible to compute the exchange difference entry and it has to be done manually.
+        currency = self[0].debit_move_id.currency_id or self[0].credit_move_id.currency_id or None
+        more_than_1_currency = False
         maxdate = '0000-00-00'
 
         seen = set()
         todo = set(self)
         while todo:
             partial_rec = todo.pop()
             seen.add(partial_rec)
-            # make sure that all partial reconciliations share the same secondary currency otherwise it's not
-            # possible to compute the exchange difference entry and it has to be done manually.
-            if currency is None and partial_rec.currency_id:
-                currency = partial_rec.currency_id
-            elif currency and partial_rec.currency_id and partial_rec.currency_id != currency:
-                # no exchange rate entry will be created
+            if partial_rec.debit_move_id.currency_id != currency or partial_rec.credit_move_id.currency_id != currency:
```

Comment:
I think we should control that partial_rec.debit_move_id.currency_id or partial_rec.credit_move_id.currency_id are not falsy